### PR TITLE
add foreign key to one to many relationship

### PR DESCRIPTION
**Pull request by rochar**

### CODE REVIEW

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs [hunk 1]
```diff
@@ -101,6 +101,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             var codeGeneratorState = new NoxSolutionCodeGeneratorState(_noxSolution, _clientAssemblyProvider.ClientAssembly);
             foreach (var entity in _noxSolution.Domain.Entities)
             {
+                Console.WriteLine($"TestWebAppDbContext Configure database for Entity {entity.Name}");
```
Comment (ILoveMyTortoise) [inline tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs:104]: do we want this?

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityZeroOrOne.cs [hunk 2]
```diff
@@ -28,4 +28,5 @@ public partial class ThirdTestEntityZeroOrOne : AuditableEntityBase
     /// ThirdTestEntityZeroOrOne Test entity relationship to ThirdTestEntityExactlyOne ZeroOrOne ThirdTestEntityExactlyOnes
     /// </summary>
     public virtual ThirdTestEntityExactlyOne? ThirdTestEntityExactlyOne { get; set; } = null!;
+    public ThirdTestEntityExactlyOne? ThirdTestEntityExactlyOneRelationship => ThirdTestEntityExactlyOne;
```
Comment (ILoveMyTortoise) [inline tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityZeroOrOne.cs:31]: maybe add space in between

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityExactlyOne.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public partial class ThirdTestEntityExactlyOne : AuditableEntityBase
     /// ThirdTestEntityExactlyOne Test entity relationship to ThirdTestEntityZeroOrOne ExactlyOne ThirdTestEntityZeroOrOnes
     /// </summary>
     public virtual ThirdTestEntityZeroOrOne ThirdTestEntityZeroOrOne { get; set; } = null!;
-
+    public ThirdTestEntityZeroOrOne ThirdTestEntityZeroOrOneRelationship => ThirdTestEntityZeroOrOne;
```
Comment (ILoveMyTortoise) [inline tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityExactlyOne.cs:31]: maybe add space in between

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs (hunk 1 above)
Comment (rochar) [reply tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs:104]: just for now... to see how workplace behaves

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs [hunk 4]
```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
...
+
+                        builder
+                            .HasOne($"{codeGeneratorState.DomainNameSpace}.{relationshipToCreate.Relationship.Entity}", relationshipToCreate.Relationship.Entity)
+                            .WithMany(entity.PluralName)
+                            .HasForeignKey($"{relationshipToCreate.Relationship.Entity}Id");
+                    }
+                    // Setup foreign key property
                     ConfigureRelationForeignKeyProperty(codeGeneratorState, builder, entity, relationshipToCreate);
-                }
+                }
```
Comment (ILoveMyTortoise) [inline src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs:99]: some weird space appeared here

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs [hunk 5]
```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
...
+                            .HasForeignKey($"{relationshipToCreate.Relationship.Entity}Id");
+                    }
+                    // Setup foreign key property
                     ConfigureRelationForeignKeyProperty(codeGeneratorState, builder, entity, relationshipToCreate);
-                }
+                }               
 
                 if (!relationshipToCreate.Relationship.ShouldUseRelationshipNameAsNavigation())
                 {
+                    Console.WriteLine($"***Ignoring Navigation {relationshipToCreate.Relationship.Name}");
```
Comment (ILoveMyTortoise) [inline src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs:103]: same, do we want to have this?

File: src/Nox.Solution/Extensions/EntityRelationshipExtensions.cs [hunk 6]
```diff
@@ -46,7 +46,7 @@ public static bool HasRelationshipWithSingularEntity(this EntityRelationship rel
 
     public static bool ShouldUseRelationshipNameAsNavigation(this EntityRelationship relationship)
     {
-        var hasReferenceToSingularEntity = HasRelationshipWithSingularEntity(relationship);
+        var hasReferenceToSingularEntity = WithSingleEntity(relationship);
```
Comment (ILoveMyTortoise) [inline src/Nox.Solution/Extensions/EntityRelationshipExtensions.cs:49]: I think bool starting with "With" is harder to understand, to be honest I like previous name more

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityZeroOrOne.cs (hunk 2 above)
Comment (rochar) [reply tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.ThirdTestEntityZeroOrOne.cs:32]: done

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs [hunk 7]
```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
...
+                if (relationshipToCreate.Relationship.ShouldGenerateForeignOnThisSide() && relationshipToCreate.Relationship.WithSingleEntity())
                 {
-                    builder
-                        .HasOne(relationshipToCreate.Relationship.Entity)
-                        .WithOne(entity.Name)
-                        .HasForeignKey(entity.Name, $"{relationshipToCreate.Relationship.Entity}Id");
-
-                    // Setup one to one foreign key
+                    //One to One
+                    if (!relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
```
Comment (ILoveMyTortoise) [inline src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs:66]: can be reversed to not have negative branch here

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs [hunk 8]
```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
...
+            EntityTypeBuilder builder,
+            Entity entity, 
+            EntityRelationshipWithType relationshipToCreate)
         {
             // Right now assuming that there is always one key present
             var key = relationshipToCreate.Relationship.Related.Entity.Keys![0];
             if (TypesDatabaseConfigurations.TryGetValue(key.Type,
                 out var databaseConfiguration))
             {
+                Console.WriteLine($"++++ConfigureRelationForeignKeyProperty {entity.Name}, " +
```
[the remark this answers is not in the document]
Comment (rochar) [reply src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs:119]: to easily identify the output :)

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs (hunk 5 above)
Comment (rochar) [reply src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs:103]: yes for now to help us troubleshoot anything, will remove before the release

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs (hunk 1 above)
Comment (ILoveMyTortoise) [reply tests/Nox.Tests/DatabaseIntegrationTests/Models/TestWebAppDbContext.cs:104]: 👍

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.TestEntityForTypes.cs [hunk 9]
```diff
@@ -94,6 +94,11 @@ public partial class TestEntityForTypes : AuditableEntityBase
     /// </summary>
     public Nox.Types.CurrencyNumber? CurrencyNumberTestField { get; set; } = null!;
 
+    /// <summary>
+    ///  (Optional).
+    /// </summary>
+    public Nox.Types.DateTime? DateTimeTestField { get; set; } = null!;
```
Comment (ILoveMyTortoise) [inline tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.TestEntityForTypes.cs:100]: hm, why has this changed? I think nothing was changes in definition

File: src/Nox.Solution/Extensions/EntityRelationshipExtensions.cs (hunk 6 above)
Comment (rochar) [reply src/Nox.Solution/Extensions/EntityRelationshipExtensions.cs:49]: that is an extension method being called like a non extension, I will change to this `relationship.WithSingleEntity();` to have the desired usage

File: tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.TestEntityForTypes.cs (hunk 9 above)
Comment (rochar) [reply tests/Nox.Tests/DatabaseIntegrationTests/Models/Entities.TestEntityForTypes.cs:100]: someone is not generating the code when changing the definition

File: src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs (hunk 7 above)
Comment (rochar) [reply src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs]: done

**Review by ILoveMyTortoise — APPROVED** [review]

**Pull request merged**